File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    answer = 'Yes'
    
    flag = True
    
    for i in goal:
        if cards1 and cards1[0] == i:
            cards1.pop(0)
        elif cards2 and cards2[0] == i:
            cards2.pop(0)
        else:
            flag = False
        
        if not flag:
            answer = "No"
            break
    
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    string answer = "Yes";
    
    bool flag = true;
    
    for (string g : goal) {
        if (!cards1.empty() && cards1[0] == g) {
            cards1.erase(cards1.begin());
        } else if (!cards2.empty() && cards2[0] == g) {
            cards2.erase(cards2.begin());
        } else {
            flag =  false;
        }
        
        if (!flag) {
            answer = "No";
            break;
        }
    }
    
    return answer;
}